rocAL/rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -119,27 +87,65 @@ def next(self):
         else:
-            self.loader.copyToTensorNHWC(self.out, self.multiplier, self.offset, self.reverse_channels, int(self.tensor_dtype))
-    
+            self.output_tensor_list =
+
+        self.augmentation_count = len(self.output_tensor_list)
+        self.w = self.output_tensor_list[0].batch_width() if self.w is None else self.w
+        self.h = self.output_tensor_list[0].batch_height() if self.h is None else self.h
+        self.batch_size = self.output_tensor_list[0].batch_size() if self.batch_size is None else self.batch_size
+        self.color_format = self.output_tensor_list[0].color_format() if self.color_format is None else self.color_format
+        
+        if self.out is None:
+            if self.tensor_layout == types.NCHW:
+                if self.device == "cpu":
+                    if self.tensor_dtype == types.FLOAT:
+                        self.out = np.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=np.float32)
+                    elif self.tensor_dtype == types.FLOAT16:
+                        self.out = np.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=np.float16)
+                    self.labels = np.empty(self.labels_size, dtype = np.int32)
+
+                else:
+                    with cp.cuda.Device(device=self.device_id):
+                        if self.tensor_dtype == types.FLOAT:
+                            self.out = cp.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=cp.float32)
+                        elif self.tensor_dtype == types.FLOAT16:
+                            self.out = cp.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=cp.float16)
+                        self.labels = cp.empty(self.labels_size, dtype = cp.int32)
+
+            else: #NHWC
+                if self.device == "cpu":
+                    if self.tensor_dtype == types.FLOAT:
+                        self.out = np.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=np.float32)
+                    elif self.tensor_dtype == types.FLOAT16:
+                        self.out = np.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=np.float16)
+                    self.labels = np.empty(self.labels_size, dtype = np.int32)
+
+                else:
+                    with cp.cuda.Device(device=self.device_id):
+                        if self.tensor_dtype == types.FLOAT:
+                            self.out = cp.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=cp.float32)
+                        elif self.tensor_dtype == types.FLOAT16:
+                            self.out = cp.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=cp.float16)
+                        self.labels = cp.empty(self.labels_size, dtype = cp.int32)
+                        
+        self.output_tensor_list[0].copy_data(ctypes.c_void_p(self.out.data_ptr()))
```

Comment:
There is a copy_data in L52
self.output_tensor_list[0].copy_data(ctypes.c_void_p(self.out.data_ptr()))
why again one more is copy_data call is present